### PR TITLE
dts: arm: st: Add support for stm32l433Xb

### PR DESCRIPTION
**Pull request by KwonTae-young**
Adds support for `stm32l433Xb`.
And add `I2C_1` and `CAN` pin of L4 series.

Signed-off-by: Kwon Tae-young <tykwon@m2i.co.kr>

### CODE REVIEW

File: drivers/pinmux/stm32/pinmux_stm32l4x.h [hunk 1]
```diff
@@ -72,11 +72,15 @@
 
 #define STM32L4X_PINMUX_FUNC_PA9_USART1_TX                                     \
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
+#define STM32L4X_PINMUX_FUNC_PA9_I2C1_SCL                                      \
+	(STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
```
Comment (FRASTM) [inline drivers/pinmux/stm32/pinmux_stm32l4x.h]: might be specific to some stm32l4xx soc devices.
can you check PA9/PA10 alternate function AF4 on L4 soc series, for example stm32l476 has no I2C1 pins on Port A

Comment (KwonTae-young) [reply drivers/pinmux/stm32/pinmux_stm32l4x.h]: The status of the L4 series I have checked is as follows.

|                 | L432 | L433 | L452 | L471 | L475 | L476 | L496 | L4r5,r9 |
| --------------- | ---- | ---- | ---- | ---- | ---- | ---- | ---- | ------- |
| PA9/PA10(I2C_1) | O    | O    | O    | X    | X    | X    | X    | X       |
| PB8/PB9(CAN)    | X    | O    | O    | O    | O    | O    | O    | O       |
| PD0/PD1(CAN)    | X    | O    | O    | O    | O    | O    | O    | O       |

I have applied the above to the source.

Comment (FRASTM) [reply drivers/pinmux/stm32/pinmux_stm32l4x.h]: However for stm32l476, that's not what is in the  [DS10198](https://www.st.com/content/ccc/resource/technical/document/datasheet/c5/ed/2f/60/aa/79/42/0b/DM00108832.pdf/files/DM00108832.pdf/jcr:content/translations/en.DM00108832.pdf)

Comment (KwonTae-young) [reply drivers/pinmux/stm32/pinmux_stm32l4x.h]: Sorry. I don't understand well.
The following code does not meet the conditions in `stm32l476`?
``` c
#if CONFIG_SOC_STM32L432XX || CONFIG_SOC_STM32L433XX || CONFIG_SOC_STM32L452XX
#define STM32L4X_PINMUX_FUNC_PA9_I2C1_SCL                                      \
	(STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
#endif

...
...

#if CONFIG_SOC_STM32L432XX || CONFIG_SOC_STM32L433XX || CONFIG_SOC_STM32L452XX
#define STM32L4X_PINMUX_FUNC_PA10_I2C1_SDA                                     \
	(STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
#endif
```

Comment (FRASTM) [reply drivers/pinmux/stm32/pinmux_stm32l4x.h]: yes, this code is _not_ valid  for stm32L476  (according to DS10198 Rev 8)

-  Alternate Function 4 for PA9 and PA10 is mapping I2C1 SCL and SDA for stm32l423, 433, 452 (but not stm32l476)

Comment (KwonTae-young) [reply drivers/pinmux/stm32/pinmux_stm32l4x.h]: `STM32L4X_PINMUX_FUNC_PA9_I2C1_SCL` is defined only when `CONFIG_SOC_STM32L432XX`, `CONFIG_SOC_STM32L433XX`, `CONFIG_SOC_STM32L452XX` are defined.
Isn't this compatible with stm32L476?

Comment (FRASTM) [reply drivers/pinmux/stm32/pinmux_stm32l4x.h]: correct: "STM32L4X_PINMUX_FUNC_PA9_I2C1_SCL is defined only when CONFIG_SOC_STM32L432XX, CONFIG_SOC_STM32L433XX, CONFIG_SOC_STM32L452XX are defined."
Maybe others but NOT stm32L476.

**Review by FRASTM — APPROVED** [review]

**Comment by ioannisg** [comment]
@erwango do you want to review this?

**Review by erwango — APPROVED** [review]

File: dts/arm/st/l4/stm32l433Xb.dtsi [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/*
```
Comment (ioannisg) [inline dts/arm/st/l4/stm32l433Xb.dtsi:1]: @KwonTae-young is this header included by any in-tree Board, SoC?

**Review by galak — APPROVED** [review]

File: dts/arm/st/l4/stm32l433Xb.dtsi (hunk 2 above)
Comment (KwonTae-young) [reply dts/arm/st/l4/stm32l433Xb.dtsi:1]: As far as I know, all *.dtsi contain these headers.

**Review by ioannisg — CHANGES_REQUESTED** [review]
@KwonTae-young you are adding support for stm32l433Xb, but the .dtsi file you introduce (dts/arm/st/l4/stm32l433Xb.dtsi) does not seem to be included anywhere in the tree, so I am wondering if this is needed for out-of-tree SoCs or Boards.

Otherwise, no objections, except for a minor request to add some proper message body in the commit that introduces the stm32l433Xb

**Comment by KwonTae-young** [comment]
@ioannisg The commit message has been changed.

File: drivers/pinmux/stm32/pinmux_stm32l4x.h [hunk 3]
```diff
@@ -72,11 +72,19 @@
 
 #define STM32L4X_PINMUX_FUNC_PA9_USART1_TX                                     \
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
+#if CONFIG_SOC_STM32L432XX || CONFIG_SOC_STM32L433XX || CONFIG_SOC_STM32L452XX
```
Comment (stephanosio) [inline drivers/pinmux/stm32/pinmux_stm32l4x.h:75]: `#if defined(CONFIG_SOC_STM32L432XX) || defined(CONFIG_SOC_STM32L433XX) || defined(CONFIG_SOC_STM32L452XX)`

File: drivers/pinmux/stm32/pinmux_stm32l4x.h [hunk 4]
```diff
@@ -72,11 +72,19 @@
...
+#define STM32L4X_PINMUX_FUNC_PA9_I2C1_SCL                                      \
+	(STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
+#endif
 
 #define STM32L4X_PINMUX_FUNC_PA10_USART1_RX                                    \
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUPDR_NO_PULL)
 #define STM32L4X_PINMUX_FUNC_PA10_OTG_FS_ID                                    \
 	(STM32_PINMUX_ALT_FUNC_10 | STM32_PUSHPULL_PULLUP)
+#if CONFIG_SOC_STM32L432XX || CONFIG_SOC_STM32L433XX || CONFIG_SOC_STM32L452XX
```
Comment (stephanosio) [inline drivers/pinmux/stm32/pinmux_stm32l4x.h:84]: `#if defined(CONFIG_SOC_STM32L432XX) || defined(CONFIG_SOC_STM32L433XX) || defined(CONFIG_SOC_STM32L452XX)`

**Review by stephanosio — CHANGES_REQUESTED** [review]

**Review by stephanosio — APPROVED** [review]

**Comment by erwango** [comment]
@ioannisg there is no STM32 dev board supporting these soc variants. So I don't think this should be blocking

**Review by ioannisg — APPROVED** [review]

**Pull request merged**